<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>542</width>
    <height>689</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Parent process</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>321</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>1.2 One process</string>
    </property>
    <widget class="QPushButton" name="pushButtonCalc">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Calc</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonExitProcess">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>20</y>
       <width>91</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Exit Process</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonTerminateProcess">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Terminate Process</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>10</y>
      <width>191</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>1.2d Three processes</string>
    </property>
    <widget class="QPushButton" name="pushButtonRun3Processes">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>151</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Run 3 processes</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>80</y>
      <width>191</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>1.3 STARTUPINFO</string>
    </property>
    <widget class="QPushButton" name="pushButtonSoliter">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>151</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Run cmd</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>321</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>1.4 Child process</string>
    </property>
    <widget class="QPushButton" name="pushButtonRunChildProc">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>291</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Run Child process</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>521</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>2 User message</string>
    </property>
    <widget class="QPushButton" name="pushButtonSendMsg">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Send message</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonPostMsg">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Post message</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEditWparam">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string notr="true"/>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEditLparam">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>60</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
     <property name="placeholderText">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>LPARAM</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>WPARAM</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonBrdCastPostMsg">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>60</y>
       <width>151</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Broadcast Post message</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonBrdCastSendMsg">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>20</y>
       <width>151</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Broadcast Send message</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>521</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>3 WM_COPYDATA</string>
    </property>
    <widget class="QLineEdit" name="lineEditCopyData">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>491</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>521</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>4.1 File mapping</string>
    </property>
    <widget class="QPushButton" name="pushButtonCreateFileMapping">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Create</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonCloseFileMapping">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Close</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelFileMappingInfo">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>110</y>
       <width>471</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>INFO:</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="textEditFileMapping">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>20</y>
       <width>271</width>
       <height>71</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonSaveFileMapping">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>70</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonLoadFileMapping">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Load</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonClearTextEdit">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>30</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>521</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>4.2 File mapping with File</string>
    </property>
    <widget class="QPushButton" name="pushButtonFMOpenFile">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="textEditFMFile">
     <property name="geometry">
      <rect>
       <x>193</x>
       <y>20</y>
       <width>311</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonFMSaveFile">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>60</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonCloseFile">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Close</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelFileMappingFInfo">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>110</y>
       <width>471</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>INFO:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonFMLoadFile">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Load</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>542</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
